lib/BlockedBy.js

```diff
@@ -44,32 +44,34 @@ class BlockedBy extends NodeResque.Plugin {
             }
         }
 
-        let waitingBuilds = await this.queueObject.connection.redis.lrange(waitingKey, 0, -1);
+        if (String(this.options.blockedBySelf) === 'true') { // only check this if feature is on
```

Comment:
can we delete the entry for the same job from delete queue ? Also since all these logic is for handing same job bock can you abstract it out to say `__handleBlockBySelf()`